homeassistant/components/mqtt/client.py

```diff
@@ -431,7 +431,7 @@ def stop():
             # Do not disconnect, we want the broker to always publish will
             self._mqttc.loop_stop()
 
-        # wait for ACK-s to be processes (unsubscribe only)
+        # wait for ACK-s to be processed (unsubscribe only)
         async with self._paho_lock:
```

Comment:
This lock should probably be applied when awaiting the tasks too. Otherwise it seems pointless here.